Lab-1 pull request

kinivi opened the pull request.

kinivi committed:
Init commit

- 1 page
- Fields for email and password
- Toggling between Sing in and Sign Up

kinivi committed:
Pre-release of lab-1

- added home screen with Sing out button
- all requested fields with validation

kinivi committed:
Gitignore update

kinivi committed:
Merge remote-tracking branch 'origin/develop' into develop

kinivi committed:
Trash delete

kinivi committed:
DS store update

kinivi committed:
Final update lab-1

ovovchak commented:
Наступні побажання - я їх озвучу на лекції завтра:
- зроби перший коміт прямо в мастер з порожнім дефолтним проектом, щоб можна було ревювати зміни повязані власне з лабораторною, а не усі генеровані файли і шукати в них написаний тобою код - щоб можна було його адекватно поревювати
- не додавай (я це вже кільком людям казав) мене чи когось іншого у ревювери, а просто добав посилання на ПР у загальну доку. За 2 роки інакше у мене буде море спаму з 80 проектів.

ovovchak commented:
Pods файли не повинні попадати в репозиторій, інакше там будуть накопичуватись зміни, яких ти реально не дизайнив...це якби ти комітив node-modules чи build папку

ovovchak commented:
symlinks чи local.properties це теж файли, яким не місце у репо, вони не мають сенсу поза твоїм лептопом

ovovchak commented:
Загальний комент на даний момент (поки важко змістовно ревювнути цей ПР) - захардкоджені стрічки, magic numbers і загальна довжина деяких функцій чи класів (по-можливості усе має бути якогомога простіше чи коротше для правильного читання)

kinivi commented:
- В доці нема просто поля для пулреквесту)
- Та, згоден там біда з сміттєвими файлами. В Flutter трохи складно з gitignore файлом
- Що мається на увазі над "захардкоджені стрічки"?

ovovchak commented:
Що мається на увазі над "захардкоджені стрічки"? - hardcoded, а саме стрінгові літерали, їх потрібно виносити у локалізаційні файли (якщо це текст для ui), або в глобальні константи(якщо це якісь параметри)

kinivi commented:
Воно так для всіх? Просто з цікавості, бо у туторіалах до Flutter як від самого Google так і взагалі ніколи не бачив щоб стрічки UI виносились в константи

https://flutter.dev/docs/development/ui/layout/tutorial

kinivi committed:
Styles file and imporvements

- added separate style file
- created api base class
- minor fixes

kinivi committed:
API and Config

- updated structure of API
- added configure files

kinivi committed:
Created FutureBuilder and Pull-to-refresh

kinivi committed:
Api and UI fixes

 - removed magical numbers
- add additional styles
- added CircullarProgress while loading
- added Signout action button

kinivi committed:
Added Snackbar
